framework/src/modules/chain/dpos/common.js

```diff
@@ -0,0 +1,48 @@
+const updateProducedBlocks = (address, undo = false) => {
```

Comment:
It needs header, and `common` is not really the name it captures this functions.